src/components/remote_control/src/commands/button_press_request.cc

```diff
@@ -0,0 +1,264 @@
+      }
+      const mobile_apis::ButtonName::eType current_button =
+          static_cast<mobile_apis::ButtonName::eType>(current_id);
+      if (current_button == button) {
+        LOG4CXX_TRACE(logger_,
+                      "Button id " << current_button
+                                   << " exist in capabilities");
+        return true;
+      }
+    }
+  }
+  LOG4CXX_TRACE(logger_,
+                "Button id " << button << " do not exist in capabilities");
+  return false;
+}
+
+bool CheckButtonName(const std::string& module_type,
+                     const std::string& button_name,
+                     const smart_objects::SmartObject* rc_capabilities) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (rc_capabilities == NULL) {
+    LOG4CXX_ERROR(logger_, "No remote controll capabilities available");
+    return false;
+  }
+
+  if (enums_value::kRadio == module_type) {
+    if (!helpers::in_range(buttons_radio(), button_name)) {
+      LOG4CXX_WARN(logger_,
+                   "Trying to acceess climate button with module type radio");
+      return false;
+    }
+  }
+
+  if (enums_value::kClimate == module_type) {
+    if (!helpers::in_range(buttons_climate(), button_name)) {
+      LOG4CXX_WARN(logger_,
+                   "Trying to acceess radio button with module type climate");
+      return false;
+    }
+  }
+  return true;
+}
+
+void ButtonPressRequest::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const Json::Value request_params =
+      MessageHelper::StringToValue(message_->json_message());
+
+  const std::string button_name = request_params[kButtonName].asString();
+  const std::string module_type = request_params[kModuleType].asString();
+
+  static ButtonsMap btn_map = buttons_map();
+  mobile_apis::ButtonName::eType button_id =
+      mobile_apis::ButtonName::INVALID_ENUM;
+  if (btn_map.end() != btn_map.find(button_name)) {
+    button_id = btn_map[button_name];
+  }
+
+  const smart_objects::SmartObject* rc_capabilities =
+      service()->GetRCCapabilities();
+  const bool button_name_matches_module_type =
+      CheckButtonName(module_type, button_name, rc_capabilities);
+  const bool button_id_exist_in_caps =
+      rc_capabilities &&
+      CheckIfButtonExistInRCCaps(*rc_capabilities, button_id);
+
+  if (button_name_matches_module_type && button_id_exist_in_caps) {
+    SendRequest(functional_modules::hmi_api::button_press, request_params);
+  } else if (!button_name_matches_module_type) {
+    LOG4CXX_WARN(logger_, "Request module type and button name mismatch!");
+    SendResponse(false,
+                 result_codes::kInvalidData,
+                 "Request module type and button name mismatch!");
+  } else {
+    LOG4CXX_WARN(logger_, "Requested button is not exists in capabilities!");
+    SendResponse(false,
+                 result_codes::kUnsupportedResource,
+                 "Requested button is not exists in capabilities!");
+  }
+}
+
+AcquireResult::eType ButtonPressRequest::AcquireResource(
```

Comment:
Are there requirement descriptions for how AcquireResource works?

Reply:
@JackLivio sure, in base_command request there is function description